### PR TITLE
Add barrierLabel in showModalBottomSheet

### LINKED ISSUE #83180: [Semantic] showModalBottomSheet with custom barrierLabel
## Proposal

Allow `showModalBottomSheet` to accept custom `barrierLabel` to make the modalSheet more descriptive for the users with ScreenReaders.

### PR DESCRIPTION
**Pull request by 838**
Allows us to pass a custom `barrierLabel` to `modalBottomSheet` to make them more descriptive for the ScreenReaders.

Fixes:  #83180 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by 838** [comment]
@googlebot I signed it!

**Comment by 838** [comment]
Hi @HansMuller
Can you please review this PR?
Thanks

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 1]
```diff
@@ -690,6 +690,7 @@ Future<T?> showModalBottomSheet<T>({
   required WidgetBuilder builder,
   Color? backgroundColor,
   double? elevation,
+  String? barrierLabel,
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/bottom_sheet.dart:693]: There should be a description of this new parameter in the dartdoc above.

**Review by darrenaustin — CHANGES_REQUESTED** [review]
Thanks for the contribution. In addition to a doc comment I mention below, we will also need a test to ensure this new parameter is working correctly (and won't be accidentally broken in the future). You can do this by adding a test to `packages/flutter/test/material/bottom_sheet_test.dart`.

**Comment by darrenaustin** [comment]
It looks like it this PR hasn't been updated in quite a while, so I am closing it for now. Feel free to reopen it again later if you want to complete this work. Thanks.

**Comment by 838** [comment]
Fixed in #126431

**Pull request closed without merging**